### ISSUE TITLE
HTMLCollection does not implement IEnumerable.

### ISSUE THREAD

**Issue by hardhub** [Retyped/Retyped#60]
Bridge.Html5 allows to use Linq on different dom collections like HTMLCollection.  

It was possible to use something like this: element.Сhildren.Select()  

What is alternative in Retyped? Or should it be considered as a bug?

**Comment by AndreyChechel** [comment]
Well, from perspective of TypeScript -> C# conversion it's not a bug. From usage perspective, especially when comparing with existing **Bridge.Html5** - it's a desired API to be in place. So I see it as enhancement. 

Anyways we'll investigate feasibility of implementing `IEnumerable` by entities declared as collections in TypeScript declaration files.

**Comment by AndreyChechel** [comment]
Confirming that collection types from at least 2 packages: **Retyped.es5** and **Retyped.dom** are to implement corresponding .NET collection interfaces. We're working on matching interfaces that should be used with a particular collection type. 

Also it's likely that collections will use explicit interface implementation to avoid mixing of C# and JavaScript APIs. LINQ extension methods and iteration using `foreach` will be available without casting, however, to access, say `ICollection.Count`, an object will need to be cast to `ICollection` type implicitly or explicitly.

**Comment by AndreyChechel** [comment]
Hi @hardhub,

The issue has been fixed, the updated packages will be available after **v1.6 release**.
Below is the list of entities and interfaces implemented by them.

**Retyped.es5**
```
Retyped.es5.Array<T>:          IList<T>, IReadOnlyList<T>, ICollection
Retyped.es5.Int8Array:         IList<sbyte>, IReadOnlyList<sbyte>, ICollection
Retyped.es5.Uint8Array:        IList<byte>, IReadOnlyList<byte>, ICollection
Retyped.es5.Uint8ClampedArray: IList<byte>, IReadOnlyList<byte>, ICollection
Retyped.es5.Int16Array:        IList<short>, IReadOnlyList<short>, ICollection
Retyped.es5.Uint16Array:       IList<ushort>, IReadOnlyList<ushort>, ICollection
Retyped.es5.Int32Array:        IList<int>, IReadOnlyList<int>, ICollection
Retyped.es5.Uint32Array:       IList<uint>, IReadOnlyList<uint>, ICollection
Retyped.es5.Float32Array:      IList<float>, IReadOnlyList<float>, ICollection
Retyped.es5.Float64Array:      IList<double>, IReadOnlyList<double>, ICollection
```

**Retyped.dom**
```
Retyped.dom.CSSRuleList:          IEnumerable<CSSRule>
Retyped.dom.DataTransferItemList: IEnumerable<DataTransferItem>
Retyped.dom.DOMRectList:          IEnumerable<DOMRect>
Retyped.dom.DOMStringList:        IEnumerable<string>
Retyped.dom.DOMTokenList:         IEnumerable<string>
Retyped.dom.FileList:             IEnumerable<File>
Retyped.dom.HTMLCollectionBase:   IEnumerable<Element>
Retyped.dom.HTMLCollectionOf<T>:  IEnumerable<T>
Retyped.dom.MediaList:            IEnumerable<string>
Retyped.dom.MimeTypeArray:        IEnumerable<MimeType>
Retyped.dom.NodeList:             IEnumerable<Node>
Retyped.dom.NodeListOf<TNode>:    IEnumerable<TNode>
Retyped.dom.PluginArray:          IEnumerable<Plugin>
Retyped.dom.StyleSheetList:       IEnumerable<StyleSheet>
Retyped.dom.SVGLengthList:        IEnumerable<SVGLength>
Retyped.dom.SVGNumberList:        IEnumerable<SVGNumber>
Retyped.dom.SVGPathSegList:       IEnumerable<SVGPathSeg>
Retyped.dom.SVGPointList:         IEnumerable<SVGPoint>
Retyped.dom.SVGStringList:        IEnumerable<string>
Retyped.dom.SVGTransformList:     IEnumerable<SVGTransform>
Retyped.dom.TextTrackList:        IEnumerable<TextTrack>
Retyped.dom.TextTrackCueList:     IEnumerable<TextTrackCue>
Retyped.dom.TouchList:            IEnumerable<Touch>
```
